src/resources/cubemap.js

```diff
@@ -170,13 +167,42 @@ Object.assign(pc, function () {
                         assetCubeMap._levelsEvents[index] = asset || null;
                     }
 
-                    // check if source is actually changed
-                    if (sources[index] !== cubemap._levels[0][index])
-                        levelsUpdated = true;
+                    var cubemap = assetCubeMap.resource;
+
+                    var levelsUpdated = false;
+                    for (var i = 0; i < 6; ++i) {
+                        if (!sources[i]) {
+                            levelsUpdated = false;
+                            break;
+                        }
+                        if (sources[i]._levels[0] !== cubemap._levels[0][i]) {
+                            levelsUpdated = true;
+                        }
+                    }
 
                     // when all faces checked
-                    if (count === 6 && levelsUpdated) {
-                        cubemap.setSource(sources);
+                    if (levelsUpdated) {
+
+                        // build levels
+                        var levels = [];
+                        for (var mip = 0; mip < sources[0]._levels.length; ++mip) {
+                            levels.push(sources.map(function (s) {
+                                // use 'this' to overcome eslint no-loop-func when referencing mip in loop
+                                return s._levels[this];
+                            }, mip));
```

Comment:
🤢 How about:

 `// eslint-disable-line no-loop-func`

Reply:
lol I tried that, didn't seem to work. I'll try again.

Reply:
worked, thanks!